CONTRIBUTING.md

```diff
@@ -1,61 +1,686 @@
-# Contributing to this repository
+# Contributing to Pipelinit
 
-Thank you for your interest in contributing to pipelinit!
+Thanks for your interest in contributing to Pipelinit. If this is your first contribution, please read our code of conduct and the Pipelinit core concepts.
 
-## Getting Started
+See below the types of contributions to learn how you can help to improve Pipelinit.
 
-Before you begin:
-
-1. Make sure you have Deno installed, see Requirements
-2. Read the [code of conduct](CODE_OF_CONDUCT.md)
-3. Read Types of contributions to learn how to contribute
 
 ## Requirements
 
-pipelinit is built with Deno, you can check how to install it
-[in the official Deno website](https://deno.land/#installation).
+* Pipelinit is built with Deno check how to install it on the official Deno website.
+* To install the executable locally, use the command:
+   * ```deno install -f --unstable --allow-read=/home/joao/ --allow-write=. cli/pipelinit.ts```
+
+
+## Types of contributions
+
+### Adding new stack plugins
+
+A Stack Plugin does two things:
+* Answers if the stack is present in the project
+* Collects data from the aforementioned stack. The data collected is what is available to templates.
```

Comment:
I think it would be useful to make a parallel here with the functions that compose a stack plugin:
```suggestion
* detect: Answers if the stack is present in the project
* introspect: Collects data from the aforementioned stack. The data collected is what is available to templates.
```